code-comment-style.md

```diff
@@ -0,0 +1,137 @@
+# Code Comment Style
+
+This document describes the code comment style applied to TiKV repositories. When you are to commit, be sure to follow the style to write good code comments.
+
+## Why a good comment matters?
+
+- To speed up the reviewing process
+- To help maintain the code
+- To improve the API document readability
+- To improve the development efficiency of the whole team
+
+## Where/When to comment?
+
+Write a comment where/when:
+
+- For important code
+- For obscure code
+- For tricky or interesting code
+- For a complex code block
+- If a bug exists in the code but you cannot fix it or you just want to ignore it for the moment
+- If the code is not optimal but you don’t have a smarter way now
+- To remind yourself or others of missing functionality or upcoming requirements not present in the code
+
+A comment is generally required for:
+
+- File
+- Type
+- Constant 
+- Function
+- Method
+- Variable
+- Typical algorithm
+- Exported name
+- Test case
+- TODO
+- FIXME
+
+## How to comment?
+
+### Format of a good comment (for Rust)
+
+- Non-doc comment
+        
+    - Use **//** for a line comment
+    - Use **/* … */** for a block comment (used only when needed)
+    
+- Doc comment
+        
+    - Use **///** for a line comment
+    - Use **//!** for a block comment
+    - Use **rustfmt** to format your code
+
+- Place the single-line and block comment above the code it’s annotating
+- Fold long lines of comments
+- Each line of text in your code and comment should be at most 100 characters long
+- Do not use relative URLs; use **absolute URLs** instead
+
+    Right:
+    
+    ```
+    // For more configuration details, see https://example.com/project/tools/config-items.  
+    ```
+    
+    Wrong:
+    
+    ```
+    // For more details, see [Configuration Items](../tools/config-items.md).
+    ```
+
+### Language for a good comment
+
+- Word
+    
+    - Use **American English** rather than British English
+        
+        - color, canceling, synchronize     (Recommended)
+        - colour, cancelling, synchronise   (Not recommended)
+    
+    - Use correct spelling
+
+    - Use **standard or official capitalization**
+        
+        - TiKV, TiDB-Binlog, Region, gRPC, RocksDB, GC, k8s, [mydumper](https://github.com/maxbube/mydumper), [Prometheus Pushgateway](https://github.com/prometheus/pushgateway)   (Right)

```

Comment:
K8s
not k8s

Reply:
https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/#what-does-kubernetes-mean-k8s

Reply:
Good spot. Let's stick with K8s in our case.

Reply:
@siddontang @dcalvin “Kubernetes” was stylized as “K8s” formerly; but now it is “k8s”.

Reply:
As later confirmed with CC, k8s in all lower-cased form is generally accepted. Only initial-capped as K8s when it is the first word of the sentence.

Reply:
As later confirmed with CC, k8s in all lower-cased form is generally accepted. Only initial-capped as K8s when it is the first word of the sentence.